modules/api/src/main/java/org/apache/ignite/table/Table.java

```diff
@@ -37,6 +39,13 @@ public interface Table {
      */
     String name();
 
+    /**
+     * Gets a partition manager of a table.
+     *
+     * @return Partition manager.
+     */
+    <T extends Partition> PartitionManager<T> partitionManager();
```

Comment:
The generic part does not make sense to me: 
* As a user, how am I supposed to know which partition type will be returned?
* As an implementer, with type erasure there is no way to know which partition type the user expects.

Reply:
As discussed in private, let's add `Class<T> partitionClass` parameter to this method:
* The user is supposed to know the partition type, because it is a part of the table schema
* If the user makes a mistake, we can throw a meaningful exception right away (instead of a `ClassCastException` somewhere later)

Reply:
Alternatively:
* Make partition classes private
* Remove generics, use `Partition` interface everywhere

Looks like the user does not need to see the actual class, they just pass it around.